src/main/java/controllers/BookstoreRestController.java

```diff
@@ -0,0 +1,177 @@
+package controllers;
+
+import models.*;
+import repositories.*;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+
+@RestController
+public class BookstoreRestController {
+
+    @Autowired
+    private BookRepository bookRepository;
+    @Autowired
+    private BookstoreRepository bookstoreRepository;
+    @Autowired
+    private BookstoreOwnerRepository bookstoreOwnerRepository;
+    @Autowired
+    private CustomerRepository customerRepository;
+    @Autowired
+    private ShoppingCartRepository shoppingCartRepository;
+    @Autowired
+    private SaleRepository saleRepository;
+
+
+    //BookstoreOwner REST endpoints
+    @GetMapping("/api/getBookstoreOwner")
+    public BookstoreOwner getBookstoreOwner(@RequestParam(value = "bookstoreOwnerId") long bookstoreOwnerId) {
+        return bookstoreOwnerRepository.findById(bookstoreOwnerId);
+    }
+
+    @GetMapping("/api/getBookstoreOwners")
+    public Iterable<BookstoreOwner> getBookstoreOwner() {
+        return bookstoreOwnerRepository.findAll();
+    }
+
+    @PostMapping("/api/newBookstoreOwner")
+    public BookstoreOwner newBookstoreOwner(@RequestParam(value = "bookstoreOwnerName") String bookstoreOwnerName) {
+        BookstoreOwner bookstoreOwner = new BookstoreOwner(bookstoreOwnerName);
+        bookstoreOwnerRepository.save(bookstoreOwner);
+        return bookstoreOwner;
+    }
+
+
+    //Bookstore REST endpoints
+    @GetMapping("/api/getBookstore")
+    public Bookstore getBookstore(@RequestParam(value = "bookstoreId") long bookstoreId) {
+        return bookstoreRepository.findById(bookstoreId);
+    }
+
+    @GetMapping("/api/getBookstores")
+    public Iterable<Bookstore> getBookstores() {
+        return bookstoreRepository.findAll();
+    }
+
+    @PostMapping("/api/newBookstore")
+    public Bookstore newBookstore(@RequestParam(value = "bookstoreName") String bookstoreName, @RequestParam(value = "bookstoreOwnerId") long bookstoreOwnerId) {
+        BookstoreOwner bookstoreOwner = bookstoreOwnerRepository.findById(bookstoreOwnerId);
+        if (bookstoreOwner == null)
+            return null;
+        Bookstore bookstore = new Bookstore(bookstoreName);
+        bookstore.setBookstoreOwner(bookstoreOwner);
+        bookstoreOwner.addBookstore(bookstore);
+        bookstoreOwnerRepository.save(bookstoreOwner);
+        return bookstore;
+    }
+
+
+    //Book REST endpoints
+    @GetMapping("/api/getBook")
+    public Book getBook(@RequestParam(value = "bookId") long bookId) {
+        return bookRepository.findById(bookId);
+    }
+
+    @GetMapping("/api/getBooks")
+    public Iterable<Book> getBooks() {
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/api/getBooksByBookstore")
```

Comment:
I believe there needs to be a few variants on this:
-  get all books for a bookstore (available and not available)
- get all available books for a bookstore
- get all non available books for a bookstore (all books sold for a bookstore)

Reply:
Addressed in dc9337a9a8da3f397032cc2d90f671df2fb05d1f